proto-lens-protoc/src/Data/ProtoLens/Compiler/Combinators.hs

```diff
@@ -259,9 +289,15 @@ tyCon = Syntax.TyCon ()
 tyList :: Type -> Type
 tyList = Syntax.TyList ()
 
+tyPromotedList :: [Type] -> Type
+tyPromotedList ts = Syntax.TyPromoted () $ Syntax.PromotedList () True ts
+
 tyPromotedString :: String -> Type
 tyPromotedString s = Syntax.TyPromoted () $ Syntax.PromotedString () s s
 
+tyPromotedBool :: Bool -> Type
```

Comment:
Style nit: specializing to `Bool` feels off for this file.  Instead:
```
type Promoted = Syntax.Promoted ()
tyPromoted :: Promoted -> Type
tyPromoted = Syntax.TyPromoted ()

instance FromString Promoted where
    fromString = Syntax.PromotedCon () True
```
and then use `tyPromoted "'True"`, etc. in Generate.

Reply:
I'm not sure I agree with this -- literally just above this is code for promoting strings. Furthermore, the strings here are actually `Type`s, meaning if we wanted to pass the buck to calling code they'd need to do something like `tyPromoted $ fromString  $ show mybool` which is pretty disgusting imo. Thoughts?